src/miral/basic_window_manager.cpp

```diff
@@ -274,11 +274,11 @@ void miral::BasicWindowManager::remove_window(Application const& application, mi
 }
 
 void miral::BasicWindowManager::refocus(
-    miral::Application const& application, miral::Window const& parent,
+    miral::Application const& application, miral::Window const& hint,
     std::vector<std::shared_ptr<Workspace>> const& workspaces_containing_window)
 {
-    // Try to make the parent active
-    if (parent && parent == select_active_window(parent))
+    // If there's a hint, try using that first
+    if (hint && hint == select_active_window(hint))
```

Comment:
```c++
if (hint && select_active_window(hint))
```

Reply:
When I made this change non-interactive layer shell surfaces were being given focus after popups were destroyed, so I had to do some more things to make it work.